src/itree.cpp

```diff
@@ -81,11 +95,25 @@ int main(int argc, char** argv){
             return 0;
         }
 
-        matcher = std::regex(vm["pattern"].as<std::string>(),
-                             std::regex::basic | std::regex::optimize );
+        auto regex_syntax_choice = std::regex::basic;
+        if (vm["regex-extended-syntax"].as<bool>()) {
+            regex_syntax_choice = std::regex::extended;
+        }
+        if (vm.count("pattern-regex")) {
+            matcher = std::regex(vm["pattern-regex"].as<std::string>(),
+                                 regex_syntax_choice | std::regex::optimize );
+        }
+        if (vm.count("pattern")) {
+            assert(is_defaulted(matcher)); // Do not use both --pattern and --pattern-regex
```

Comment:
What message is printed when the assertion fails?

The error message needs to have the same feeling as other iRODS messages.

Reply:
I'll look around for some examples of more fitting iRODS messages in other icommands, then.  I'm not sure what format the others use when it's not a message from the server.

Reply:
Sure. The main issue is that the message produced by a failing assertion is too developer oriented and doesn't provide enough information. See output of godbolt example: https://gcc.godbolt.org/z/jn8zM398W

A better message would be something like `Incompatible options: --flag-1 and --flag-2 cannot be used together.`